lib/buildingsync/model_articulation/facility.rb

```diff
@@ -60,50 +60,49 @@ def initialize(facility_xml, standard_to_be_used, ns)
 
     # adding a site to the facility
     def read_xml(facility_xml, standard_to_be_used, ns)
-      # puts facility_xml.to_a
       facility_xml.elements.each("#{ns}:Sites/#{ns}:Site") do |site_element|
         @sites.push(Site.new(site_element, standard_to_be_used, ns))
       end
     end
 
     # generating the OpenStudio model based on the imported BuildingSync Data
-    def generate_baseline_osm(epw_file_path, standard_to_be_used)
+    def generate_baseline_osm(epw_file_path, output_path, standard_to_be_used)
       if @sites.count == 0
         OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Facility.generate_baseline_osm', 'There are no sites attached to this facility in your BuildingSync file.')
         raise 'There are no sites attached to this facility in your BuildingSync file.'
-      else if @sites.count > 1
-             OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Facility.generate_baseline_osm', "There are more than one (#{@sites.count}) sites attached to this facility in your BuildingSync file.")
-             raise "There are more than one (#{@sites.count}) sites attached to this facility in your BuildingSync file."
-           else
-             OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.generate_baseline_osm', "Info: There is/are #{@sites.count} sites in this facility.")
-           end
+      elsif @sites.count > 1
+        OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Facility.generate_baseline_osm', "There are more than one (#{@sites.count}) sites attached to this facility in your BuildingSync file.")
+        raise "There are more than one (#{@sites.count}) sites attached to this facility in your BuildingSync file."
+      else
+        OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.generate_baseline_osm', "Info: There is/are #{@sites.count} sites in this facility.")
       end
       @sites[0].generate_baseline_osm(epw_file_path, standard_to_be_used)
 
-      create_building_systems(@sites[0].get_model, @sites[0].get_building_template, @sites[0].get_system_type, @sites[0].get_climate_zone, 'Forced Air')
+      create_building_systems(output_path)
+      return true
     end
 
-    def create_building_systems(model, template, system_type, climate_zone, hvac_delivery_type)
-      add_space_type_loads = true
-      add_constructions = true
-      add_elevators = false
-      add_exterior_lights = false
+    def get_sites
+      return @sites
+    end
+
+    def create_building_systems(output_path, hvac_delivery_type = 'Forced Air', htg_src = 'NaturalGas', clg_src = 'Electricity',
```

Comment:
we need to run `rubocop --auto-correct` on this before this is merged.

Reply:
done